### PR TITLE
swift 5.5

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Plus:

* Strip debug symbols from some binaries on Linux (makes Linux bottle relocatable)
* Patch a resource dir issue on Linux where Swift-compiled programs would have a runpath pointing to the Cellar
* Improved arm64e fallback (upstream patch)

### CODE REVIEW

**Comment by SMillerDev** [comment]
They seem to have retagged libdispatch?

**Comment by Bo98** [comment]
Seems so. It looks like they might have switched branches (there's two 5.5 release branches - one for the initial 5.5 release and one for 5.5.x) for some reason.

**Comment by Bo98** [comment]
Ok this seems to work now except the test for some reason takes a tad too long on 10.15 (not sure why - it probably will even pass on the next run).

File: Formula/swift.rb [hunk 1]
```diff
@@ -84,64 +84,77 @@ class Swift < Formula
...
+    sha256 "5210ca0fd95b960d596c058f5ac76412a6987d2badf5394856bb9e31d3c68833"
+  end
+
+  # Fix shims and clang headers not being copied when stdlib isn't built.
   patch do
-    url "https://github.com/apple/swift/commit/86c40574f594f4f7b4b25bb02cc2389e1328c200.patch?full_index=1"
-    sha256 "6d22d159b8c16ec14c4ec1e7ea258ffa489f932842b0f4be930d0f9660cbe9b4"
+    url "https://gist.githubusercontent.com/Bo98/33a4eac264b0571b9f4a248e87b37af7/raw/1b1978bc685480dcb393808ba980d937f1e74c81/swift-shims-all.diff"
+    sha256 "0e2c2dc96895931e4444f90533cb2b8e1b04c2591f9a2c0492145661efab1760"
   end
```
Comment (Bo98) [inline Formula/swift.rb:152]: I'm upstreaming this patch.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**